public/app/features/canvas/runtime/scene.tsx

```diff
@@ -375,10 +411,88 @@ export class Scene {
     });
   };
 
+  contextMenuAction = (actionType: string) => {
+    this.selection.pipe(first()).subscribe((currentSelectedElements) => {
+      const currentSelectedElement = currentSelectedElements[0];
+      const currentLayer = currentSelectedElement.parent!;
+
+      switch (actionType) {
+        case LayerActionID.Delete:
+          currentLayer.doAction(LayerActionID.Delete, currentSelectedElement);
+          break;
+        case LayerActionID.Duplicate:
+          currentLayer.doAction(LayerActionID.Duplicate, currentSelectedElement);
+          break;
+        case LayerActionID.MoveTop:
+          currentLayer.doAction(LayerActionID.MoveTop, currentSelectedElement);
+          break;
+        case LayerActionID.MoveBottom:
+          currentLayer.doAction(LayerActionID.MoveBottom, currentSelectedElement);
+          break;
+      }
+    });
+  };
+
+  renderMenuItems = () => {
+    return (
+      <>
+        <MenuItem
+          label="Delete"
+          onClick={(e) => {
+            this.contextMenuAction(LayerActionID.Delete);
+            this.closeContextMenu();
+          }}
+          className={this.styles.menuItem}
+        />
+        <MenuItem
+          label="Duplicate"
+          onClick={(e) => {
+            this.contextMenuAction(LayerActionID.Duplicate);
+            this.closeContextMenu();
+          }}
+          className={this.styles.menuItem}
+        />
+        <MenuItem
+          label="Bring to front"
+          onClick={(e) => {
+            this.contextMenuAction(LayerActionID.MoveTop);
+            this.closeContextMenu();
+          }}
+          className={this.styles.menuItem}
+        />
+        <MenuItem
+          label="Send to back"
+          onClick={(e) => {
+            this.contextMenuAction(LayerActionID.MoveBottom);
+            this.closeContextMenu();
+          }}
+          className={this.styles.menuItem}
+        />
+      </>
+    );
+  };
+
+  closeContextMenu = () => {
+    this.contextMenu.isMenuVisible = false;
+    this.showContextMenu.next(false);
+  };
+
+  renderContextMenu = () => {
+    return (
+      <ContextMenu
+        x={this.contextMenu.anchorPoint.x}
+        y={this.contextMenu.anchorPoint.y}
+        onClose={this.closeContextMenu}
+        renderMenuItems={this.renderMenuItems}
+      />
+    );
+  };
+
   render() {
     return (
       <div key={this.revId} className={this.styles.wrap} style={this.style} ref={this.setRef}>
         {this.root.render()}
+        {this.contextMenu.isMenuVisible && this.renderContextMenu()}
```

Comment:
I would recommend rendering the menu inside a Portal to make sure it is rendered above everything, and can extend the bounds of the panel